lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosAlgorithm.cxx

```diff
@@ -338,7 +338,12 @@ void FunctionalChaosAlgorithm::run()
     Psi_k.add(basis.build(i));
   }
   // Build the result
-  result_ = FunctionalChaosResult(inputSample_, outputSample_, distribution_, transformation_, inverseTransformation_, basis, I_k, alpha_k, Psi_k, residuals, relativeErrors);
+  result_ = FunctionalChaosResult(inputSample_, outputSample_, distribution_, transformation_,
+                                  inverseTransformation_, basis, I_k, alpha_k, Psi_k,
+                                  residuals, relativeErrors);
+  result_.setIsLeastSquares(projectionStrategy_.isLeastSquares());
+  result_.setInvolvesModelSelection(adaptiveStrategy_.getImplementation()->involvesModelSelection() || 
```

Comment:
Sad to have to check both the adaptive strategy and the projection strategy when only the adaptive strategy should be checked...

Reply:
This is because `LeastSquaresMetaModelSelectionFactory` is a projection strategy. That is how the current code is.